### ISSUE TITLE
There is no way currently to send HTTPS request in Hyper

### ISSUE THREAD

**Issue by e-oz** [hyperium/hyper#952]
To send https request, one need Hyper, compiled with feature "ssl". It requires rust-openssl and rust-openssl-verify. rust-openssl dropped support of previous versions, so we can use only 0.9. openssl-verify 0.1 is still points to rust-openssl 0.7.17. It could be fixed by vendoring, but rust-openssl-verify 0.2 is incompatible with rust-openssl 0.9. Awesome. And security-framework only works on OS X.
It looks like issues of other libraries, but in result I don't understand how to send HTTPS request in Hyper.

**Comment by e-oz** [comment]
rust-openssl-verify has been declared deprecated just now. It's the reason to update dependencies :)

**Comment by pimeys** [comment]
I'm having the same problem and just figuring out how much work it would be for me to branch this and update the dependencies. Was not that complex with solicit, but here I already struggle a bit...

**Comment by e-oz** [comment]
I spent around 13 hours on it and in result reverted to previous version.

**Comment by pimeys** [comment]
Have you tried the master or tokio branches? I could refactor my code to use the async version, but how stable are those?

**Comment by e-oz** [comment]
I've tried master and it has incompatibilities.

**Comment by seanmonstar** [comment]
I'm confused. It should still be possible to send requests over HTTPS with v0.9.11. Just because a new version of openssl has released doesn't mean latest version of hyper stops working.

**Comment by e-oz** [comment]
How it's happened in my case: after some update (maybe OpenSSL update on the ec2 instance, maybe new build of my project), openssl 0.7.14 started crashing on attempts to create ssl connection.
After that I tried a lot of combinations to create Cargo.lock with only one version of openssl, but Hyper isn't compatible with openssl 0.9.

To make things even better, ssh2 published new version with minor (last number) change, and despite I had strict "0.2.13" in Cargo.toml, Cargo decided it's not critical update and it would be nice to have new shiny version. With dependency on openssl 0.9 :) So now each time I increase version of app in Cargo.toml, cargo build fails, then I manually change version of ssh2 to 0.2.13 in Cargo.lock and only after that cargo build works.

So there is a choice: or you stick with current version of Hyper, openssl 0.7 and old versions of other libraries, dependent on openssl, or you upgrade other libraries but... Hyper will not be able to send HTTPS.

**Comment by seanmonstar** [comment]
@e-oz interesting. I spoke to Alex, since upgrading the `-sys` crate dependency is mostly a breaking change, and he agreed. There is now ssh2 at 0.2.15 which uses openssl-sys 0.7, and he'll publish a 0.3 with the new openssl.

**Comment by e-oz** [comment]
Thanks to @seanmonstar there is a way now: https://github.com/hyperium/hyper/issues/949#issuecomment-259576662
And I think it's the better way.
